Trading/Mode/index_trading_mode/index_trading.py

```diff
@@ -198,25 +237,96 @@ async def _send_alert_notification(self):
+                    f"{coin} (holdings: {round(coin_ratio * trading_constants.ONE_HUNDRED, 3)}%) is not in index "
+                    f"anymore. A rebalance is required."
+                )
+        # compute coins to buy or sell
         for coin in self.trading_mode.indexed_coins:
             coin_ratio = self.exchange_manager.exchange_personal_data.portfolio_manager. \
                 portfolio_value_holder.get_holdings_ratio(coin, traded_symbols_only=True)
             target_ratio = self.trading_mode.get_target_ratio(coin)
-            if (coin_ratio == trading_constants.ZERO and target_ratio > trading_constants.ZERO) or (
-                not (
-                    target_ratio - self.trading_mode.rebalance_cap_ratio
-                    < coin_ratio
-                    < target_ratio + self.trading_mode.rebalance_cap_ratio
-                )
-            ):
+            beyond_ratio = True
+            if coin_ratio == trading_constants.ZERO and target_ratio > trading_constants.ZERO:
+                # missing coin in portfolio
+                rebalance_details[RebalanceDetails.ADD.value][coin] = target_ratio
+                should_rebalance = True
+            elif coin_ratio < target_ratio - self.trading_mode.rebalance_cap_ratio:
+                # not enough in portfolio
+                rebalance_details[RebalanceDetails.BUY_MORE.value][coin] = target_ratio
+                should_rebalance = True
+            elif coin_ratio > target_ratio + self.trading_mode.rebalance_cap_ratio:
+                # too much in portfolio
+                rebalance_details[RebalanceDetails.SELL_SOME.value][coin] = target_ratio
+                should_rebalance = True
+            else:
+                beyond_ratio = False
+            if beyond_ratio:
                 self.logger.info(
-                    f"{coin} is beyond the target ratio of {target_ratio * trading_constants.ONE_HUNDRED}%, "
-                    f"ratio = {coin_ratio * trading_constants.ONE_HUNDRED}%. A rebalance is required"
+                    f"{coin} is beyond the target ratio of {round(target_ratio * trading_constants.ONE_HUNDRED, 3)}%, "
+                    f"ratio: {round(coin_ratio * trading_constants.ONE_HUNDRED, 3)}%. A rebalance is required."
                 )
-                return True
-        return False
+        self._resolve_swaps(rebalance_details)
+        for origin, target in rebalance_details[RebalanceDetails.SWAP.value].items():
+            origin_ratio = round(
+                rebalance_details[RebalanceDetails.REMOVE.value][origin] * trading_constants.ONE_HUNDRED,
+                3
+            )
+            target_ratio = round(
+                rebalance_details[RebalanceDetails.ADD.value].get(
+                    target,
+                    rebalance_details[RebalanceDetails.BUY_MORE.value].get(target, trading_constants.ZERO)
+                ) * trading_constants.ONE_HUNDRED,
+                3
+            ) or "???"
```

Comment:
Is this value only used for logging purpose ?

Reply:
yep